openssl-ts: add page (#6125)
tldr-pages/tldr: pages/common/openssl-ts.md

# openssl ts

> OpenSSL command to generate and verify timestamps.
> More information: <https://www.openssl.org/docs/manmaster/man1/openssl-ts.html>.

> - Generate sha512 timestamp request of `file` (output to `file.tsq`):

Reviewer: ```suggestion
- Generate SHA-512 timestamp request of `file` (output to `file.tsq`):
```

> - Generate SHA-512 timestamp request of `file` (output to `file.tsq`):

Reviewer 2: ```suggestion
- Generate a SHA-512 timestamp request of `file` (output to `file.tsq`):
```

> - Generate a SHA-512 timestamp request of `file` (output to `file.tsq`):

Reviewer: ```suggestion
- Generate a SHA-512 timestamp request of a specific file and output to `file.tsq`:
```

> `openssl ts -query -data {{file}}  -sha512  -out {{file.tsq}}`

Reviewer: ```suggestion
`openssl ts -query -data {{path/to/file}}  -sha512  -out {{path/to/file.tsq}}`
```

Author: Why do some tldr pages (eg openssl, gpg) use simple filenames like {{public.gpg}}, whereas others use {{path/to/file}}? Is there a convention based on the context?

Author: Thank you so much for reviewing my pull request ^_^

> - Check timestamp response file `file.tsr`:

Reviewer: I don't like this sentence @bl-ue Any suggestions?

Author: How about this?
- Check date and metadata of timestamp response file `file.tsr`:

Reviewer: I like that you are inputing, my problem is the `file file` bit

Author: Lol, I completely missed that.

Author: - Check date and metadata from `file.tsr`, the timestamp response file:

How about this?

Reviewer: Well I haven't used this tool, so I don't understand the context but that doesn't read right to me

- Check the date and metadata from a specified timestamp response file

Author: Are you sure it's a good idea not to reference that the timestamp response file is the argument with the .tsr extension?

You submit the timestamp request file (.tsq) to the time stamp server, it returns a time stamp request file (.tsr). This shows the date on the tsr and the metadata, but it doesn't verify the server's signature.

Here is a sample output for the command:

Using configuration from /usr/lib/ssl/openssl.cnf
Status info:
Status: Granted.
Status description: unspecified
Failure info: unspecified

TST info:
Version: 1
Policy OID: tsa_policy1
Hash Algorithm: sha512
Message data:
    0000 - cf 83 e1 35 7e ef b8 bd-f1 54 28 50 d6 6d 80 07   ...5~....T(P.m..
    0010 - d6 20 e4 05 0b 57 15 dc-83 f4 a9 21 d3 6c e9 ce   . ...W.....!.l..
    0020 - 47 d0 d1 3c 5d 85 f2 b0-ff 83 18 d2 87 7e ec 2f   G..<]........~./
    0030 - 63 b9 31 bd 47 41 7a 81-a5 38 32 7a f9 27 da 3e   c.1.GAz..82z.'.>
Serial number: 0x9826A4
Time stamp: Jun 14 21:59:39 2021 GMT
Accuracy: unspecified
Ordering: yes
Nonce: unspecified
TSA: DirName:/O=Free TSA/OU=TSA/description=This certificate digitally signs documents and time stamp requests made using the freetsa.org online services/CN=www.freetsa.org/emailAddress=busilezas@gmail.com/L=Wuerzburg/C=DE/ST=Bayern
Extensions:

Reviewer: My suggestion uses it, and the file extension specifies it in the command

Author: Ok, I implemented your suggestion.

> - Check the date and metadata from a specified timestamp response file:

Reviewer 2: ```suggestion
- Check the date and metadata of a specific timestamp response file:
```

> `openssl ts -reply -in {{file.tsr}} -text`

Reviewer: ```suggestion
`openssl ts -reply -in {{path/to/file.tsr}} -text`
```

> - Verify timestamp request `file.tsq` and timestamp response `file.tsr` from server with ssl certificate `cert.pem`:

Reviewer 2: ```suggestion
- Verify timestamp request `file.tsq` and timestamp response `file.tsr` from the server with ssl certificate `cert.pem`:
```

> - Verify timestamp request `file.tsq` and timestamp response `file.tsr` from the server with ssl certificate `cert.pem`:

Reviewer: ```suggestion
- Verify a timestamp request file and a timestamp response file from the server with ssl certificate file:
```

Reviewer 2: nit: `ssl` → `an SSL`

> `openssl ts -verify -in {{file.tsr}} -queryfile {{file.tsq}} -partial_chain -CAfile {{cert.pem}}`

Reviewer: ```suggestion
`openssl ts -verify -in {{path/to/file.tsr}} -queryfile {{path/to/file.tsq}} -partial_chain -CAfile {{path/to/cert.pem}}`
```

> - Create time stamp response for request `file.tsq` using key `tsakey.pem` and signing certificate `tsacert.pem` (output to `file.tsr`):

Reviewer 2: ```suggestion
- Create a timestamp response for request `file.tsq` using key `tsakey.pem` and signing certificate `tsacert.pem` (output to `file.tsr`):
```

> - Create a timestamp response for request `file.tsq` using key `tsakey.pem` and signing certificate `tsacert.pem` (output to `file.tsr`):

Reviewer: ```suggestion
- Create a timestamp response for request using key and signing certificate and output to `file.tsr`:
```
Leave this one for others

Reviewer 2: I'd say
```suggestion
- Create a timestamp response for request using key and signing certificate, and output it to `file.tsr`:
```

Author: `- Create a timestamp response for request using key and signing certificate, and output the timestamp to `file.tsr`:`

What about?

Reviewer 2: sgtm 👍🏻

Reviewer: the `it` is required however the comma isn't needed

Author: > the `it` is required however the comma isn't needed

So
` - Create a timestamp response for request using key and signing certificate and output it to `file.tsr`:`

Reviewer: Yes

Reviewer 2: >the `it` is required however the comma isn't needed

I don't really agree, but @0x1080's [latest](https://github.com/tldr-pages/tldr/pull/6125#discussion_r652132016) looks good to me; let's go with it.

> `openssl ts -reply -queryfile {{file.tsq}} -inkey {{tsakey.pem}} -signer tsacert.pem -out {{file.tsr}}`

Reviewer: ```suggestion
`openssl ts -reply -queryfile {{path/to/file.tsq}} -inkey {{path/to/tsakey.pem}} -signer tsacert.pem -out {{path/to/file.tsr}}`
```
